### ISSUE TITLE
Under QEMU 6.1 sway does not recognize virtio-gpu as a hardware renderer

### ISSUE THREAD

**Issue by wkral** [NixOS/nixpkgs#144320]
### Describe the bug
When running with QEMU 6.1 a VM created a few versions ago will fail to run sway without explicitly telling it to use the software renderer which has significatly degraded performance.

Sway debug errors when running under QEMU 6.1:

```
00:00:00.039 [ERROR] [wlr] [render/egl.c:281] Software rendering detected, please use the WLR_RENDERER_ALLOW_SOFTWARE environment variable to proceed
00:00:00.039 [ERROR] [wlr] [EGL] command: eglMakeCurrent, error: EGL_BAD_DISPLAY (0x3008), message: "Invalid display (nil)"
00:00:00.039 [ERROR] [wlr] [render/gles2/renderer.c:742] Could not initialize EGL
```

I have isolated it to QEMU, by adding an overlay of QEMU prior to commit 988da51d9c4, and sway runs normally.

### Steps To Reproduce

1. Create a VM with sway with QEMU <= 6.0, observe it running with virtualized hardware acceleration
2. Upgrade to 6.1.0
3. Run sway inside VM with QEMU 6.1.0, and observe the crash.

Params used to run the VM:

```
qemu-kvm \
    -smp 4 \
    -m 12G \
    -vga virtio \
    -display gtk,gl=on \
    -nic user,model=virtio-net-pci \
    -audiodev pa,id=snd0 \
    -device ich9-intel-hda \
    -device hda-output,audiodev=snd0 \
    -device qemu-xhci,id=xhci \
    -device usb-tablet \
    -device usb-host,bus=xhci.0,vendorid=0x0bda,productid=0x58b0 \
    -drive file=test.qcow2,if=virtio \
```

Guest and host are both NixOS.

### Expected behavior

Sway should run with hardware accelleration in EGL context.

### Additional context
There is a nixos test for sway which essentially runs in a similar setup which passes on Hydra builds. It would seem to be showing a false positive with this context. Changing the parameters to use `-vga none -device virtio-gpu-pci` as the test does, for me has the same failure.

There was also an upgrade to libvirtio in a4d1cc9c23042fb0ee11e1313d16a888d35b9056 but my overlay is working with that new version so I don't think that is the cause of the problem.

### Notify maintainers

@alyssais

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11pre326932.b165ce0c4ef"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by alyssais** [comment]
This is https://gitlab.com/qemu-project/qemu/-/issues/586.  @wkral, does adding "-device virtio-vga-gl" as mentioned in that issue fix the problem for you?

**Comment by wkral** [comment]
@alyssais Yes it does. Thanks very much, I searched for quite a while and wasn't able to find that issue, or any other doc mentioning the change.

**Comment by wkral** [comment]
I'm sorry I didn't test this correctly. I unknowingly had software rendering enabled in my VM for the test I did yesterday. When I went to use it today performance really degradded, I didn't notice at first but it seems that it got worse over time. Curiously, I think I must have had software rendering enabled in the last week or so while I had reverted QEMU 6.0 and performance wasn't so bad. Or possibly sway ignores the software rendering flag if it has access to hardware rendering.

I have since tested sway hardware rendering with the `-device virtio-vga-gl` flag in QEMU 6.1, but it fails with the same error.

I also moved back to the 6.0 overlay and re-tested without the software rendering flag, and that works correctly.
